### DISCUSSION TITLE
Run binary commands from dependencies of dependencies

### DISCUSSION THREAD

**Discussion by domoritz** [yarnpkg/berry#2716] [Q&A]
_Originally posted by @bkarlson in https://github.com/yarnpkg/berry/issues/2188#issuecomment-753995168_

I ran into this issue after an hour of searching. It would be good to get at least some warning if there is an indirect dependency that exposes the command. 

Now, if a direct dependency wants to expose a bin command from a dependency, how would it do that?

**Comment by arcanis** [comment]
> Now, if a direct dependency wants to expose a bin command from a dependency, how would it do that?

It provides its own binary that forwards the call to whatever other binary it wants. There is no particular builtin support for that.

**Reply by domoritz** [reply]
Thanks for moving this to a discussion. From your response, I read that there is no way in yarn to explicitly call a binary from a subdependency, right? This means I need to update the dependency (which is okay in my case). 

What's the cleanest way to forward binary command that does not break compatibility with npm/yarn 1?
